### PR TITLE
feat: provide boundary support for agent modules

### LINKED ISSUE #457: Integrate coder/boundary with AI modules
We want the AI modules in the registry to have the option to run with coder/boundary in front.

### PR DESCRIPTION
**Pull request by shanewhite97**
## Description
Enable any agent module to run its AI agent inside Coder's Agent Boundaries.
The agentapi module handles boundary installation, config setup, and wrapper
script creation, then exports AGENTAPI_BOUNDARY_PREFIX for consuming modules
to use in their start scripts.

Supports three boundary installation modes:
- coder boundary subcommand (default, Coder v2.30+)
- Standalone binary via install script (use_boundary_directly)
- Compiled from source (compile_boundary_from_source)

Users must provide a boundary config.yaml with their allowlist and
settings when enabling boundary.

Closes #457

## Type of Change
- [x] Feature/enhancement

## Module Information
**Path:** `registry/coder/modules/agentapi`
**Breaking change:** No

## Testing & Validation
- [x] Tests pass (`bun test`)
- [x] Code formatted (`bun fmt`)
- [x] Changes tested locally

### CODE REVIEW

File: registry/coder/modules/agentapi/main.test.ts [hunk 1]
```diff
@@ -612,5 +623,133 @@ describe("agentapi", async () => {
...
+      ]);
+      expect(configCheck.stdout.trim()).toBe("missing");
+      // BOUNDARY_WRAPPER should NOT be in the mock log
+      const mockLog = await readFileContainer(
+        id,
+        "/home/coder/agentapi-mock.log",
+      );
+      expect(mockLog).not.toContain("BOUNDARY_WRAPPER:");
+    });
+    test("boundary-enabled-generates-config", async () => {
```
Comment (zedkipp) [inline registry/coder/modules/agentapi/main.test.ts]: I think we could make a more comprehensive test that proves agent subprocess was actually launched through the "boundary" wrapper if we updated `testdata/agentapi-start.sh` to splice in the wrapper. Then the mock binary can check if it was invoked with `coder boundary -- <cmd>`.

This would prove the full chain: main.sh sets wrapper env var -> start script uses it -> agent subprocess runs through `coder boundary -- <cmd>`. The current test only exercises the first link in that chain.

It would also serve as a reference implementation for future module devs.

File: registry/coder/modules/agentapi/scripts/main.sh [hunk 2]
```diff
@@ -109,9 +113,50 @@ export LC_ALL=en_US.UTF-8
 
 cd "${WORKDIR}"
 
+# Set up boundary if enabled
+export BOUNDARY_WRAPPER=""
```
[the remark this answers is not in the document]
Comment (shanewhite97) [reply registry/coder/modules/agentapi/scripts/main.sh]: Done — renamed to AGENTAPI_BOUNDARY_PREFIX throughout (main.sh, agentapi-start.sh, agentapi-mock.js, and tests). This also makes it clear to module authors where the variable originates.

File: registry/coder/modules/agentapi/README.md [hunk 3]
```diff
@@ -89,6 +89,41 @@ module "agentapi" {
...
+Boundary requires a `config.yaml` file with your allowlist, jail type, proxy port, and log
+level. See the [Agent Boundaries documentation](https://coder.com/docs/ai-coder/agent-boundaries)
+for configuration details.
+To enable:
+
+```tf
+module "agentapi" {
+  # ... other config
+  enable_boundary      = true
+  boundary_config_path = "${path.module}/boundary-config.yaml"
```
[the remark this answers is not in the document]
Comment (shanewhite97) [reply registry/coder/modules/agentapi/README.md]: This is really good Zed thanks for pointing this out. Changed from boundary_config_path (file path) to boundary_config (file contents). The content is now base64-encoded at plan time and decoded inside the workspace, following the same pattern as install_script, start_script, and pre_install_script. The README example now uses file("${path.module}/boundary-config.yaml") to read the config at plan time.
Hopefully this is as you envisioned it 😄

File: registry/coder/modules/agentapi/main.tf [hunk 4]
```diff
@@ -164,6 +164,22 @@ variable "module_dir_name" {
...
+variable "boundary_config" {
+  type        = string
+  description = "Content of boundary config.yaml file. Required when enable_boundary is true. Must contain allowlist, jail_type, proxy_port, and log_level."
+  default     = ""
+  validation {
+    condition     = !var.enable_boundary || var.boundary_config != ""
+    error_message = "boundary_config is required when enable_boundary is true."
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (shanewhite97) [reply registry/coder/modules/agentapi/main.tf]: @35C4n0r These changes have now been made. Let me know if there is anything else I need to wrap up 😄

File: registry/coder/modules/agentapi/scripts/boundary.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    make build
+
+    sudo cp boundary /usr/local/bin/
+    sudo chmod +x /usr/local/bin/boundary
+    cd - || exit 1
+  elif [ "${USE_BOUNDARY_DIRECTLY}" = "true" ]; then
+    echo "Installing boundary using official install script (version: ${BOUNDARY_VERSION})"
+    curl -fsSL https://raw.githubusercontent.com/coder/boundary/main/install.sh | bash -s -- --version "${BOUNDARY_VERSION}"
+  else
+    echo "Using coder boundary subcommand (provided by Coder)"
```
[the remark this answers is not in the document]
Comment (35C4n0r) [reply registry/coder/modules/agentapi/scripts/boundary.sh]: @shanewhite97 let's add this function to the boundary.sh

```
validate_boundary_subcommand() {
  if command_exists coder; then
    if coder boundary --help > /dev/null 2>&1; then
      return 0
    else
      echo "Error: 'coder' command found but does not support 'boundary' subcommand. Please enable install_boundary or install a compatible version of Coder." >&2
      exit 1
    fi
  else
    echo "Error: ENABLE_BOUNDARY=true, but 'coder' command not found. Boundary cannot be enabled." >&2
    exit 1
  fi
}
```

File: registry/coder/modules/agentapi/scripts/boundary.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+#!/usr/bin/env bash
+set -euo pipefail
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+exec "${SCRIPT_DIR}/coder-no-caps" boundary -- "$@"
+WRAPPER_EOF
+    else
+      echo "Error: ENABLE_BOUNDARY=true, but 'coder' command not found. Boundary cannot be enabled." >&2
+      exit 1
+    fi
+  fi
```
Comment (35C4n0r) [inline registry/coder/modules/agentapi/scripts/boundary.sh]:
```suggestion
    CODER_NO_CAPS="$module_path/coder-no-caps"
    if ! cp "$(which coder)" "$CODER_NO_CAPS"; then
      echo "Error: Failed to copy coder binary to ${CODER_NO_CAPS}. Boundary cannot be enabled." >&2
      exit 1
    fi
    cat > "${BOUNDARY_WRAPPER_SCRIPT}" << 'WRAPPER_EOF'
#!/usr/bin/env bash
set -euo pipefail
SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
exec "${SCRIPT_DIR}/coder-no-caps" boundary -- "$@"
WRAPPER_EOF
  fi
```

**Review by 35C4n0r — CHANGES_REQUESTED** [review]

File: registry/coder/modules/agentapi/scripts/boundary.sh (hunk 6 above)
Comment (35C4n0r) [reply registry/coder/modules/agentapi/scripts/boundary.sh]: Helps resolve: https://github.com/coder/registry/pull/780#discussion_r2907555903
